video_xblock/video_xblock.py

```diff
@@ -773,6 +817,67 @@ def get_transcript_download_link(self):
                 return transcript.get('url')
         return ''
 
+    def create_transcript_file(self, ext='.vtt', trans_str='', reference_name=''):
+        """
+        Upload a transcript, fetched from a video platform's API, to video xblock.
+        Arguments:
+            ext (str): format of transcript file, default is vtt.
+            trans_str (str): multiple string for convert to vtt file.
+            reference_name (str): name of transcript file.
+        Returns:
+            File's file_name and external_url.
+        """
+
+        # Define location of default transcript as a future asset and prepare content to store in assets
+        file_name = reference_name.replace(" ", "_") + ext
+        course_key = self.location.course_key  # pylint: disable=no-member
+        content_loc = self.static_content.compute_location(course_key, file_name)  # AssetLocator object
+        content = self.static_content(
+            content_loc,
+            file_name,
+            'application/json',
+            trans_str.encode('UTF-8')
+        )  # StaticContent object
+        external_url = '/' + str(content_loc)
+
+        # Commit the content
+        self.contentstore().save(content)
+
+        return file_name, external_url
+
+    def convert_3playmedia_caps_to_vtt(self, caps, video_id, lang="en", lang_label="English"):
+        """
+        Utility method to convert any supported transcripts into WebVTT format.
+
+        Arguments:
+            caps (unicode)  : Raw transcripts.
+            video_id (str)  : Video id from player.
+            lang (str)      : Iso code for language.
+            lang_label (str): Name of language.
+        Returns:
+            response (dict) : {"lang": lang, "url": url, "label": lang_label}
+                lang (str)  : Iso code for language.
+                url (str)   : External url for vtt file.
+                label (str) : Name of language.
+        """
+        out = []
+        for item in caps.splitlines():
+            if item == '':
+                item = ' \n'
+            elif '-->' in item:
+                item = item[:29]
```

Comment:
>>> len('00:05:55.030 --> 00:05:57.200')
29
I think this way better then regexp, because if we don't have valid time string vtt converter raise exception.